### DISCUSSION TITLE
How to create a website section?

### DISCUSSION THREAD

**Discussion by fourclever** [mastodon/mastodon#26371] [Q&A]
#### I just started learning Mastodon and unfortunately I don't know Ruby and React. As the first goal, I chose to create my section in the public interface. The existing "About" section was taken as a sample. By analogy, I did:

1. Created about2/index.js in app/javascript/mastodon/features/ with the same content as in about/index.js

2. Added to app/javascript/mastodon/features/ui/util/async-components.js
`export function About2 () {
  return import(/*webpackChunkName: "features/about2" */'../../about2');
}`

3. Added to app/javascript/mastodon/features/ui/index.js
`import { ...
  About2, ... 
} from './util/async-components';`

4. Added to config/routes.rb
`get '/about2', to: 'about2#show'`

5. Created spec/controllers/about2_controller_spec.rb with code:
`require 'rails_helper'
RSpec.describe About2Controller, type: :controller do
...`

6. Created app/controllers/about2_controller.rb with code:
`class About2Controller < ApplicationController
...`

7. Created app/views/about2/show.html.haml with code exactly as in app/views/about/show.html.haml

#### But I get an error 404. Please tell me what else is missing?

**Comment by mjankowski** [comment]
Did you ever sort this out?

Closing since its a year old and more of a rails/react dev question than a thing about mastodon specifically.

My suggestion would be look at the server logs and see whats happening.

**Reply by trankten** [reply]
Hello, just for the record.

I was able to create a section with fully support for registered users at https://tkz.one/extras

If anyone needs support on this contact me on Mastodon at @trankten@tkz.one

Best regards,
